### PR TITLE
Fix UnionDoc type hint missing in init_beanie and on FindInterface

### PR DESCRIPTION
**Pull request by CAPITAINMARVEL**

### CODE REVIEW

File: beanie/odm/interfaces/find.py [hunk 1]
```diff
@@ -27,10 +27,15 @@
 
 if TYPE_CHECKING:
     from beanie.odm.documents import Document
+    from beanie.odm.union_doc import UnionDoc
     from beanie.odm.views import View
 
+    FindType = TypeVar("FindType", bound=Union["Document", "UnionDoc", "View"])
+else:
+    FindType = TypeVar("FindType", bound=Union["Document", "View"])
```
Comment (adeelsohailahmed) [inline beanie/odm/interfaces/find.py:35]: Please correct me if I'm wrong, but aren't both of these types intended for type checking?

**Review by adeelsohailahmed — COMMENTED** [review]
Thank you for the changes! I'll give it a more thorough review soon.

File: beanie/odm/utils/init.py [hunk 2]
```diff
@@ -76,7 +78,7 @@ def __init__(
 
         :param database: AsyncIOMotorDatabase - motor database instance
         :param connection_string: str - MongoDB connection string
-        :param document_models: List[Union[Type[DocType], str]] - model classes
+        :param document_models: List[Union[Type[DocType], Type[UnionDocType] str]] - model classes
```
Comment (adeelsohailahmed) [inline beanie/odm/utils/init.py:81]:
```suggestion
        :param document_models: List[Union[Type[DocType], Type[UnionDocType], str]] - model classes
```
Missing a comma here.

**Review by adeelsohailahmed — COMMENTED** [review]
Looks good to me.

**Review by 07pepa — APPROVED** [review]

**Review by ayusshsoarav — APPROVED** [review]

**Pull request merged**